### DISCUSSION TITLE
Show last command output in pager binding

### DISCUSSION THREAD

**Discussion by siaal** [ghostty-org/ghostty#2968] [Feature Requests, Ideas] [answered]
Probably my favourite kitty feature is the ability to grab some amount of scrollback buffer and open it in a pager.
Usually used with either the entire scrollback buffer, or just the buffer since the last prompt.

It's extremely convenient for searching through output when you didn't have the foresight to pipe it into a pager manually, and it preserves colour codes that would usually be stripped when pagers were detected.

For reference in kitty the features are called `show_scrollback` (<C+S+h>) and `show_last_command_output` (<C+S+g>) and seem to be implemented via
`less --chop-long-lines --RAW-CONTROL-CHARS +INPUT_LINE_NUMBER`

**Comment by ralgozino** [comment]
I know this is not exactly what you are asking for, but there's already a keybinding to dump the scrollback buffer to a file and another one to open the scrollback buffer in your text editor:

```bash
$ ghostty +list-keybinds | grep scrollback
keybind = super+alt+shift+j=write_scrollback_file:open
keybind = super+shift+j=write_scrollback_file:paste
```

There's been some discussion on the search scrollback feature thread about these kitty features too:
- https://github.com/ghostty-org/ghostty/issues/189

**Reply by siaal** [reply]
I did see <c-s-j>, but it would vomit out the entire scrollback, most of the time I only want to look through the last command (<c-s-g> in kitty).

My original round of searching didn't see #189, thanks for that!  That discussion seems fairly focused on the search behaviour though, while I'm fairly married to the idea of throwing it into a buffer so I can navigate it with vim motions.

**Reply by xzbdmw** [reply]
It seems `write_scrollback_file:paste` only write invisible area's text.

**Reply by 00-kat** [reply, marked as answer]
@xzbdmw very late, but `write_screen_file`. #3496 #3652.

**Reply by stefanvanburen** [reply]
I see this is closed, but I'd still like to see a separate command for retrieving _just_ the last command's output (`show_last_command_output` from kitty suggested above). This can be helpful when e.g. you're running tests with a lot of output, and you only want to look through the latest run of the tests.

**Reply by 00-kat** [reply]
Related: #8797 and #6810.

**Comment by mbtuckersimm** [comment]
This is a really useful feature of kitty and this is probably the main thing that's preventing me from switching over to ghostty. I know that you can open the scrollback buffer in a text editor or dump it to a file, but those actions both require you to switch contexts. It's really useful to be able to open the output of the last command right where you are, scroll through to find what you're looking for, and then dismiss it with another keypress.
